api/base/utils.py

```diff
@@ -151,8 +152,8 @@ def default_node_permission_queryset(user, model_cls):
     assert model_cls in {Node, Registration}
     if user.is_anonymous:
         return model_cls.objects.filter(is_public=True)
-    sub_qs = Contributor.objects.filter(node=OuterRef('pk'), user__id=user.id, read=True)
-    return model_cls.objects.annotate(contrib=Exists(sub_qs)).filter(Q(contrib=True) | Q(is_public=True))
+    read_user_query = Q(id__in=get_objects_for_user(user, 'read_node', model_cls))
```

Comment:
Some drive-by nits:

* Would be good to pass `with_superuser=False`
* You can use `values_list('id')` here